maestro-driver-ios/maestro-driver-iosUITests/Routes/Handlers/SubTreeRouteHandler.swift

```diff
@@ -12,27 +12,30 @@ final class SubTreeRouteHandler: HTTPHandler {
             logger.error("Requested view hierarchy for an invalid appId")
             return HTTPResponse(statusCode: HTTPStatusCode.badRequest)
         }
-
+        
         do {
             let springboardBundleId = "com.apple.springboard"
             logger.info("Trying to capture hierarchy snapshot for \(appId)")
             let start = NSDate().timeIntervalSince1970 * 1000
             let xcuiApplication = XCUIApplication(bundleIdentifier: appId)
             let springboardApplication = XCUIApplication(bundleIdentifier: springboardBundleId)
-
+            
+            tapOnSystemPermissionAlertIfNeeded(springboardApplication: springboardApplication,
+                                               appName: xcuiApplication.label)
+            
```

Comment:
Instead of mixing this into the implementation of SubTree, can the permission code go into its own file. Then each handler could first call into the permission handler and then continue with its work.
Later we could then refactor it as a pre-action for every handler so that the handlers are free from permission logic.

Reply:
do you think it makes sense to do it on every Handler or just subTree?

Reply:
resolved